### PR TITLE
improve testing regarding expired keys and keys with only email addresses

### PR DESCRIPTION
**Pull request by joshrabinowitz**
For #238 and #281 . 
~~Code in progress~~
This code is ready for merge.

### CODE REVIEW

**Comment by joshrabinowitz** [comment]
@julianStier @adamgotterer : 

Current master will issue an erorr if any file failed to decrypt (which technically fixes #238 'Git secret ignoring gpg issues with expired public key') but it won't give you any hint why.

I'm thinking of adding a patch to git-secret that informs users about keys that expire. 
My first thought is to always mention the expiration date of any keys with expiration dates. 
Any suggestions, thoughts or input on what git-secret should do about keys that have expiration times?

**Comment by adamgotterer** [comment]
Awesome! Thanks for adding!

`whoknows` could also list the expirations or days until expiration by default or with a flag to show/hide. Could also have a  flag for "expiring soon".

If `hide` tells you it failed because there was an expired key(s) and lists the offender keys there could be a follow up prompt that 'Y' offers to remove those keys and continue encryption and 'N' cancels encryption.

**Comment by rasendubi** [comment]
I experience a similar issue with an expired uid when I issue `git secret hide`.

git-secret calls the following command:
```sh
gpg --homedir=<repo>/.gitsecret/keys --no-permission-warning --use-agent --yes --trust-model=always --encrypt -r<uid1> -r<uid2_expired> -r<uid3_expired> -r<uid4> -o <repo>/<file>.secret <repo>/<file>
```

(I was able to fix the issue by manually `gpg --homedir=<repo>/.gitsecret/keys --edit-key <uid1>` and deleting my expired uids.)

Does this PR fixes that? I don't see any changes in `git_secret_hide.sh`. That's why I'm asking.

**Comment by joshrabinowitz** [comment]
this PR is in progress.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## {{Next Version}}
+
+- Add tests for expired gpg keys, and gpg keys with only emails.
+
```
Comment (joshrabinowitz) [inline CHANGELOG.md]: Is it ok if we put in change information for the next changelog this way?

Comment (sobolevn) [reply CHANGELOG.md]: Yes, I do the same thing.

Comment (joshrabinowitz) [reply CHANGELOG.md]: excellent, thank you

**Pull request merged**